kong/clustering/services/config.lua

```diff
@@ -0,0 +1,80 @@
+local utils = require("kong.tools.utils")
+local cjson = require("cjson.safe")
+
+local ngx = ngx
+local ngx_time = ngx.time
+local ngx_log = ngx.log
+local INFO = ngx.INFO
+local assert = assert
+local inflate_gzip = utils.inflate_gzip
+local cjson_decode = cjson.decode
+local yield = utils.yield
+
+local _M = {}
+
+local _log_prefix = "[wrpc-clustering] "
+
+-- we should move funcitions about config sync here but that would be too large a refactoring.
+
+---- CP part
+
+local function init_config_cp(wrpc_service)
+  wrpc_service:import("kong.services.config.v1.config")
+
+  wrpc_service:set_handler("ConfigService.PingCP", function(peer, data)
+    local client = peer.client
+    if client and client.update_sync_status then
+      client.last_seen = ngx_time()
+      client.config_hash = data.hash
+      client:update_sync_status()
+      ngx_log(INFO, _log_prefix, "received ping frame from data plane")
+    end
+  end)
+
+  wrpc_service:set_handler("ConfigService.ReportMetadata", function(peer, data)
+    local client = peer.client
+    if client then
+      ngx_log(INFO, _log_prefix, "received initial metadata package from client: ", client.dp_id)
+      client.basic_info = data
+      client.basic_info_semaphore:post()
+    end
+    return {
```

Comment:
`{ok = 'done'}` should be a constant table

Reply:
Yes. That's a remaining from the last PR.